packages/superinput/scripts/genDocTree.ts

```diff
@@ -0,0 +1,180 @@
+
+const recurseGenerateTree: RecurseGenerateTreeFn = (node, entries) => {
+  Object.entries(entries).forEach(entry => {
+    const [key, value] = entry
+    let tmpNode: DocNode = tree.parse({
+      key,
+      type: null,
+      link: null,
+      docs: null
+    })
+    if (typeof value === 'object' && Object.keys(value).length > 0) {
+      //@ts-ignore
+      const { description, type, $ref, enum: enumm } = value
+      if (description) {
+        tmpNode.model.link = extractLink(description)
+        tmpNode.model.docs = cleanDocs(description)
+      }
+
+      if (enumm) {
+        tmpNode.model.type = `Enum(${enumm.join(',')})`
+      } else if (type) {
+        switch (type) {
+          case 'array':
+            const items = value?.items as TJS.Definition
+            if (items.$ref) {
+              tmpNode.model.type = `${snipRef(items.$ref)}[]`
+            } else {
+              tmpNode.model.type = type
+            }
+            break
+          case 'object':
+            const { properties, additionalProperties } = value
+            tmpNode.model.type = type
+            if (properties) {
+              tmpNode = recurseGenerateTree(tmpNode, properties)
+            }
+            if (additionalProperties && (additionalProperties as TJS.Definition).$ref) {
+              const snippedRef = snipRef((additionalProperties as TJS.Definition).$ref)
+              const tmpDesc = (definitions[snippedRef] as TJS.Definition).description
+              tmpNode.model.type = `{key: ${snippedRef}}`
+              tmpNode.addChild(
+                tree.parse({
+                  key: '*',
+                  type: snippedRef,
+                  link: extractLink(tmpDesc),
+                  desc: cleanDocs(tmpDesc)
+                })
+              )
+            }
+            break
+          default:
+            tmpNode.model.type = type
+        }
+      } else if ($ref && typeof definitions[snipRef($ref)] === 'object') {
+        tmpNode.model.type = snipRef($ref)
+        tmpNode = recurseGenerateTree(tmpNode, (definitions[snipRef($ref)] as TJS.Definition).properties || {})
+      }
+    }
+    node.addChild(tmpNode)
+  })
+  return node
+}
+
+function recurseGenerateFallback(node: any): any {
+  let val: any = {}
+  if (node.children) {
+    val = node.children.reduce((accu: any, child: DocNode) => {
+      if (child.children) {
+        accu[child.key] = recurseGenerateFallback(child)
+      } else {
+        accu[child.key] = null
+      }
+
+      return accu
+    }, {})
+  }
+  return val
+}
+
+const docTree = recurseGenerateTree(root, properties).model
+let fallback: any = recurseGenerateFallback(docTree)
+
+fallback = {
+  event: fallback,
+  ...fallback,
+  ...fallback.state
+}
+
+const docs = {
+  docTree,
+  fallback
+}
+
+fs.writeFileSync(join(__dirname, '../src/docsTree.json'), JSON.stringify(docs))
+// eslint-disable-next-line
+console.log(`Generated doctree at ${__dirname + '../src/docsTree.json'}`)
```

Comment:
I added the console log so it can be a little verbose when this step is done, the eslint comment is because we don't allow console.log.

I can remove it if its not needed.